test/e2e/scheduling/preemption.go

```diff
@@ -516,3 +562,19 @@ func waitForPreemptingWithTimeout(f *framework.Framework, pod *v1.Pod, timeout t
 	})
 	framework.ExpectNoError(err, "pod %v/%v failed to preempt other pods", pod.Namespace, pod.Name)
 }
+
+func getCurrentPodUsageOnTheNode(nodeName string, pods []v1.Pod, resource *v1.ResourceRequirements) (int64, int64) {
+	totalRequestedCPUResource := resource.Requests.Cpu().MilliValue()
+	totalRequestedMemResource := resource.Requests.Memory().Value()
+	for _, pod := range pods {
+		if pod.Spec.NodeName == nodeName {
+			if v1qos.GetPodQOS(&pod) == v1.PodQOSBestEffort {
+				continue
+			}
+		}
+		result := getNonZeroRequests(&pod)
```

Comment:
This doesn't take into account init containers, so there's a small chance that this fails because a larger init container than regular container is running.  I think it's ok for now, but we should fix this.  Resources used for a pod is `max(max(init_container usage), sum(container usage))`

Reply:
Why non-zero requests? Why not just ignore resource requests which are not set?

Reply:
Sorry, what is the difference between them? It seems like this function is essentially just trying to get the requests, not caring about zero (or unset) because they don't add into the usage. Is that right?

Reply:
`GetNonzeroRequests` which is called by `getNonZeroRequests` and implemented in https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/util/non_zero.go adds +100m or 200MB in case cpu/memory resource requests is not set. So `getNonZeroRequests` can give you higher usage on a node that it is.